extensions/schema-compare/src/models/schemaCompareOptionsModel.ts

```diff
@@ -0,0 +1,1381 @@
+		loc.IgnoreDefaultSchema,
+		loc.IgnoreDdlTriggerState,
+		loc.IgnoreDdlTriggerOrder,
+		loc.IgnoreCryptographicProviderFilePath,
+		loc.VerifyDeployment,
+		loc.IgnoreComments,
+		loc.IgnoreColumnCollation,
+		loc.IgnoreAuthorizer,
+		loc.IgnoreAnsiNulls,
+		loc.GenerateSmartDefaults,
+		loc.DropStatisticsNotInSource,
+		loc.DropRoleMembersNotInSource,
+		loc.DropPermissionsNotInSource,
+		loc.DropObjectsNotInSource,
+		loc.IgnoreColumnOrder,
+	].sort();
+
+	public objectTypeLabels: string[] = [
+		loc.Aggregates,
+		loc.ApplicationRoles,
+		loc.Assemblies,
+		loc.AssemblyFiles,
+		loc.AsymmetricKeys,
+		loc.BrokerPriorities,
+		loc.Certificates,
+		loc.ColumnEncryptionKeys,
+		loc.ColumnMasterKeys,
+		loc.Contracts,
+		loc.DatabaseOptions,
+		loc.DatabaseRoles,
+		loc.DatabaseTriggers,
+		loc.Defaults,
+		loc.ExtendedProperties,
+		loc.ExternalDataSources,
+		loc.ExternalFileFormats,
+		loc.ExternalTables,
+		loc.Filegroups,
+		loc.Files,
+		loc.FileTables,
+		loc.FullTextCatalogs,
+		loc.FullTextStoplists,
+		loc.MessageTypes,
+		loc.PartitionFunctions,
+		loc.PartitionSchemes,
+		loc.Permissions,
+		loc.Queues,
+		loc.RemoteServiceBindings,
+		loc.RoleMembership,
+		loc.Rules,
+		loc.ScalarValuedFunctions,
+		loc.SearchPropertyLists,
+		loc.SecurityPolicies,
+		loc.Sequences,
+		loc.Services,
+		loc.Signatures,
+		loc.StoredProcedures,
+		loc.SymmetricKeys,
+		loc.Synonyms,
+		loc.Tables,
+		loc.TableValuedFunctions,
+		loc.UserDefinedDataTypes,
+		loc.UserDefinedTableTypes,
+		loc.ClrUserDefinedTypes,
+		loc.Users,
+		loc.Views,
+		loc.XmlSchemaCollections,
+		loc.Audits,
+		loc.Credentials,
+		loc.CryptographicProviders,
+		loc.DatabaseAuditSpecifications,
+		loc.DatabaseEncryptionKeys,
+		loc.DatabaseScopedCredentials,
+		loc.Endpoints,
+		loc.ErrorMessages,
+		loc.EventNotifications,
+		loc.EventSessions,
+		loc.LinkedServerLogins,
+		loc.LinkedServers,
+		loc.Logins,
+		loc.MasterKeys,
+		loc.Routes,
+		loc.ServerAuditSpecifications,
+		loc.ServerRoleMembership,
+		loc.ServerRoles,
+		loc.ServerTriggers
+	].sort();
+
+	constructor(defaultOptions: mssql.DeploymentOptions) {
+		this.deploymentOptions = defaultOptions;
+	}
+
+	public getOptionsData(): string[][] {
+		let data = [];
+		this.optionsLookup = {};
+		this.optionsLabels.forEach(l => {
+			let checked: boolean = this.GetSchemaCompareOptionUtil(l);
+			data.push([checked, l]);
+			this.optionsLookup[l] = checked;
+		});
+		return data;
+	}
+
+	public getObjectsData(): string[][] {
+		let data = [];
+		this.objectsLookup = {};
+		this.objectTypeLabels.forEach(l => {
+			let checked: boolean = this.GetSchemaCompareIncludedObjectsUtil(l);
+			data.push([checked, l]);
+			this.objectsLookup[l] = checked;
+		});
+		return data;
+	}
+
+	public SetDeploymentOptions() {
+		for (let option in this.optionsLookup) {
+			this.SetSchemaCompareOptionUtil(option, this.optionsLookup[option]);
+		}
+	}
+
+	public SetSchemaCompareOptionUtil(label: string, value: boolean) {
+		switch (label) {
```

Comment:
Why is this a massive switch case?

Reply:
this isn't a 1 to 1 mapping, since we're going from the localized label to the variable being set, and then setting that to the value. Is there an easy way to change that to something simpler? This PR was also mostly meant to just move the existing code to another file, so I'd probably do a more involved change like that in a separate PR later.

Reply:
I didn't really see anywhere where a label was changing two settings, but maybe even special casing those would potentially fit this better?